lib/ae_mdw/sync/server.ex

```diff
@@ -311,21 +313,61 @@ defmodule AeMdw.Sync.Server do
       end)
 
     if MapSet.size(accounts_set) > 0 do
-      {block_index, mb_hash, mblock} = last_microblock(gen_mutations)
-
-      account_balances_mutations =
-        mb_hash
-        |> get_balances_from_node(accounts_set)
-        |> Enum.reduce([], fn {account_pk, balance}, acc ->
-          [{block_index, mblock, UpdateBalanceAccountMutation.new(account_pk, balance)} | acc]
-        end)
-
-      gen_mutations ++ account_balances_mutations
+      gen_mutations
+      |> last_microblock()
+      |> case do
+        {:ok, {block_index, mb_hash, _mblock}} ->
+          account_balances_mutations =
+            mb_hash
+            |> get_balances_from_node(accounts_set)
+            |> Enum.reduce([], fn {account_pk, balance}, acc ->
+              [UpdateBalanceAccountMutation.new(account_pk, balance) | acc]
+            end)
+
+          update_gen_mutations(gen_mutations, block_index, account_balances_mutations)
+
+        {:error, :no_mblocks} ->
+          gen_mutations
+      end
```

Comment:
Why did this change?

Reply:
The last_microblock function was trying to get the max element of an empty list and was crashing the sync process

Reply:
Let's move these changes to a separate PR and mark it as "fix: ..."